src/azure-cli/azure/cli/command_modules/appservice/_validators.py

```diff
@@ -85,3 +85,27 @@ def validate_app_exists_in_rg(cmd, namespace):
             app = client.web_apps.get(resource_group_name, webapp, None, raw=True)
         if app.response.status_code != 200:
             raise CLIError(app.response.text)
+
+
+def validate_add_vnet(cmd, namespace):
+    if isinstance(namespace.name, str) and isinstance(namespace.resource_group, str):
+        resource_group_name = namespace.resource_group
+        from azure.cli.command_modules.network._client_factory import network_client_factory
+        vnet_client = network_client_factory(cmd.cli_ctx)
+        list_all_vnets = vnet_client.virtual_networks.list_all()
+        vnet = namespace.vnet
+        name = namespace.name
+        slot = namespace.slot
+
+        vnet_loc = ''
+        for v in list_all_vnets:
+            if v.name == vnet:
+                vnet_loc = v.location
```

Comment:
I would cross check whether or not the service end would return a good error. If yes, we can remove the client end validations; otherwise you need to make sure you compare with the full resource id and also `break` once you find a match